### ISSUE TITLE
irods 4.0.3 hardcodes home directory path on install, but tries to use $HOME for irods user account

### ISSUE THREAD

**Issue by kript** [irods/irods#2354]
Hi Folks,

On a clean install of 4.0.3 I ran into an issue caused by a conflict between our environment and the irods installation expectations.

Installing the system went fine, but I got the following error after running setup;

```
Step 5 of 5:  Configuring iRODS user and starting server...
    Updating iRODS user's ~/.irods/.irodsEnv...
    Starting iRODS server...
    Opening iRODS connection...

Install problem:
    Connection to iRODS server failed.
        ERROR: Save Password failure status = -900000 FILE_OPEN_ERR
Found 2 processes:
Stopping process id 11103
Stopping process id 11104
Killing process id 11103
Killing process id 11104

Abort.
```

This turned out to be because the installation directory was /var/lib/irods but the service uses the $HOME directory of the irods user, which in our case is different, since the irods user is in our directory service (LDAP).

I symlinked that directory (which it wasn't able to create as the irods user) to the installation directory, and tried to start up iRODS. That failed with the following in rodsLog;

```
Sep 30 14:55:34 pid:11251 ERROR: queZone:  Bad input portNum 0 for 
    LocalHostName:  localhost, isg-dev-1-09, isg-dev-1-09.internal.sanger.ac.uk,Sep 30 15:09:30 pid:11687 NOTICE: created irodsHome=
Sep 30 15:09:30 pid:11687 ERROR: queZone:  Bad input portNum 0 for 
    LocalHostName:  localhost, isg-dev-1-09, isg-dev-1-09.internal.sanger.ac.uk,Sep 30 15:12:14 pid:11939 NOTICE: environment variable set, irodsEnvFile=/var/lib/irods/iRODS/conf
ig/irodsEnv.boot
```

Which some googling showed me that the .irodsEnv file hadn't been created properly, and it wasn;t able to get the right port number.  Thinking would be part of the setup_irods.sh I reran the setup script.

This ran OK but ended with some error messages;

```
Step 5 of 5:  Configuring iRODS user and starting server...
    Updating iRODS user's ~/.irods/.irodsEnv...
    Starting iRODS server...
    Opening iRODS connection...
    Creating default resource...
        ... Success [demoResc] [/var/lib/irods/iRODS/Vault]
    Testing resource...

Done.  Additional detailed information is in the log file:
    /var/lib/irods/iRODS/installLogs/irods_setup.log

-----------------------------
Running update_catalog_schema.py...
Traceback (most recent call last):
  File "./packaging/update_catalog_schema.py", line 214, in <module>
    main()
  File "./packaging/update_catalog_schema.py", line 209, in main
    update_database_to_latest_version()
  File "./packaging/update_catalog_schema.py", line 154, in update_database_to_latest_version
    current_schema_version = get_current_schema_version(cfg)
  File "./packaging/update_catalog_schema.py", line 82, in get_current_schema_version
    result[1].decode('utf-8').split("\n")[resultline].strip())
ValueError: invalid literal for int() with base 10: ‘OPTION_VALUE'
```

I appear to be able to iput/ils etc, but perhaps this is an artifact of running the setup twice against Oracle?

In summary two issues, I think;
1. The irods user needs to be able to handle a different path for its home directory from the installation binaries.
2. Is the Oracle portion of the setup script idempotent?

cheers

John
p.s. not all bad news, the Oracle plugin worked fine!

**Comment by trel** [comment]
closing this now - we have incorporated most of this knowledge/experience into the dbplugin 1.4 backport [#2466]
